src/Service.Tests/Configuration/HotReload/ConfigurationHotReloadTests.cs

```diff
@@ -299,4 +299,36 @@ public async Task HotReloadConfigRuntimeGQLEnabledEndToEndTest()
         // Assert
         Assert.AreEqual(HttpStatusCode.NotFound, gQLResult.StatusCode);
     }
+
+    /// <summary>
+    /// Hot reload the configuration file by saving a new database type and connection string.
+    /// Validate that the response from the server is correct when making a new request after
+    /// the change in database type.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow(DatabaseType.MySQL, TestCategory.MYSQL, DisplayName = "Hot-reload change to MySQL Datasource")]
+    [DataRow(DatabaseType.PostgreSQL, TestCategory.POSTGRESQL, DisplayName = "Hot-reload change to PostgreSQL Datasource")]
+    public async Task HotReloadConfigDatasourceDatabaseTypesEndToEndTest(DatabaseType expectedDatabaseType, string expectedTestCategory)
+    {
+        // Arrange
+        string expectedConnectionString = $"{ConfigurationTests.GetConnectionStringFromEnvironmentConfig(expectedTestCategory).Replace("\\", "\\\\")}";
+
+        GenerateConfigFile(
+            databaseType: expectedDatabaseType,
+            connectionString: expectedConnectionString);
+        System.Threading.Thread.Sleep(3000);
+
+        // Act
+        RuntimeConfig updatedRuntimeConfig = _configProvider.GetConfig();
```

Comment:
it might be useful to retrieve what the database type was before the hot reload, then assert it was not the same as the expected one - because the idea would be to test changing the db type..